arm/Microsoft.OperationalInsights/workspaces/.bicep/nested_solutions.bicep

```diff
@@ -0,0 +1,21 @@
+param gallerySolution string
+param logAnalyticsWorkspaceName string
+param location string
+
+resource logAnalyticsWorkspace 'Microsoft.OperationalInsights/workspaces@2020-08-01' existing = {
+  name: logAnalyticsWorkspaceName
+}
+
+resource solution 'Microsoft.OperationsManagement/solutions@2015-11-01-preview' = {
+  name: '${gallerySolution}(${logAnalyticsWorkspace.name})'
+  location: location
+  properties: {
+    workspaceResourceId: logAnalyticsWorkspace.id
+  }
+  plan: {
+    name: '${gallerySolution}(${logAnalyticsWorkspace.name})'
+    product: 'OMSGallery/${gallerySolution}'
+    promotionCode: ''
+    publisher: 'Microsoft'
```

Comment:
I'd avoid hardcoded values for `product `and `publisher` in the resource section. I'd suggest to add them as parameters with default values instead

Reply:
done :)